pkg/products/threescale/reconciler.go

```diff
@@ -3540,3 +3551,196 @@ func (r *Reconciler) reconcileServiceMonitor(ctx context.Context, client k8sclie
 
 	return integreatlyv1alpha1.PhaseCompleted, nil
 }
+
+func (r *Reconciler) reconcileDcEnvarEmailAddress(ctx context.Context, serverClient k8sclient.Client, dcName string, updateFn func(dc *appsv1.DeploymentConfig, value string) bool) (integreatlyv1alpha1.StatusPhase, error) {
+	observabilityConfig, err := r.ConfigManager.ReadObservability()
+	if err != nil {
+		return integreatlyv1alpha1.PhaseFailed, err
+	}
+
+	existingSMTPFromAddress, err := resources.GetSMTPFromAddress(ctx, serverClient, r.log, r.installation, observabilityConfig.GetNamespace())
+	if err != nil {
+		return integreatlyv1alpha1.PhaseFailed, err
+	}
+
+	dc, err := r.appsv1Client.DeploymentConfigs(r.Config.GetNamespace()).Get(ctx, dcName, metav1.GetOptions{})
+	if err != nil {
+		return integreatlyv1alpha1.PhaseFailed, err
+	}
+
+	updated := updateFn(dc, existingSMTPFromAddress)
+
+	if updated {
+		dc, err = r.appsv1Client.DeploymentConfigs(dc.Namespace).Update(ctx, dc, metav1.UpdateOptions{})
+		if err != nil {
+			return integreatlyv1alpha1.PhaseFailed, err
+		}
+		err = r.RolloutDeployment(ctx, dcName)
+		if err != nil {
+			r.log.Error(fmt.Sprintf("Rollout %v deployment", dcName), err)
+			return integreatlyv1alpha1.PhaseFailed, err
+		}
+
```

Comment:
From what I have experienced it does need to the called. There is also the following example of this happening else.
https://github.com/integr8ly/integreatly-operator/blob/master/pkg/products/threescale/reconciler.go#L2366

Reply:
Oh okay, though I think these rollout's are done when underlying resources used such as the config maps and secrets are updated in order for the pods to take in the changes 🤔 

I know when the DCs are updated to include the envoy labels / annotations typically causes a rollout without needing to manually do a rollout, so thought it might be the case here aswell 🤔

Reply:
I though so as well but on less it was very delayed it didn't the rollout.  These feature is used in a few different spots. 

What is the ask here?

Reply:
Ah nothing, was just wondering. Conversation can be resolved 👍